src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DocumentSnapshot.cs

```diff
@@ -12,8 +14,16 @@ internal abstract class DocumentSnapshot
 
         public abstract string TargetPath { get; }
 
+        public abstract Task<SourceText> GetTextAsync();
+
+        public abstract Task<VersionStamp> GetTextVersionAsync();
+
         public abstract Task<RazorCodeDocument> GetGeneratedOutputAsync();
 
+        public abstract bool TryGetText(out SourceText result);
+
+        public abstract bool TryGetTextVersion(out VersionStamp result);
+
         public abstract bool TryGetGeneratedOutput(out RazorCodeDocument result);
```

Comment:
Just an idea. Would this benefit from tracking `Imports` as well? Seems like something that would go with the document.

Reply:
Yep. I figured that if there was someone else helping me with this, that might be a good thing for him to take on.

Reply:
Then it's a good thing that someone is already on it. 👍